### ISSUE TITLE
Customisation of Serialisation (DateTime)

### ISSUE THREAD

**Issue by TehWardy** [OData/WebApi#959]
The idiot that came before me somehow managed to fudge the data in to the db with all dates in utc time despite being the SQL DateTime type, yet when my code gets hold of it with EF and then hands it over to OData the offset is applied "by some means" that matches the servers localtime.
This means all my DateTimes come out an hour out (as the server is in France, 2 hours during DST)

I'm not suggesting there's anything wrong with OData here, but I would like to request a means to "manually serialise" a property of a given type using a convention.
Is there some way / would it be possible to add a "Property serlialisation rule" that given a property value of a given type could handle returning that value as a string.

I recently started using EF Dynamic Filters (https://github.com/jcachat/EntityFramework.DynamicFilters), in that we can apply a filter to a type in the model using something like ...

`builder.Filter("MyFilter", (T t) => t.Foo == "somevalue");`

I wondered if it might be possible to do something like ...

`builder.OnSerialising<T>(t => { return "my sutom output for the value t"; });`

### Assemblies affected
Microsoft.AspNet.OData 5.9.1
Microsoft.OData.Core 6.15
Microsoft.OData.Edm 6.15

### Reproduce steps
Any standard query on OData endpoints

### Expected result
Custom output for anything of type T based on custom serialisation rule added to model

### Actual result
OData determined serialisation output that I can't control directly

**Comment by TehWardy** [comment]
On second thoughts I might be looking at this wrong.
This situation came about because we have a database that stores DateTime values in GMT on a server setup in another time zone, the net result is that OData "adjusts" our values when it does whatever its doing to expose the property as a DateTimeOffset in the model (to fit OData standards of course)... 

I think a simpler fix maybe to have some sort of option that allows me to specify the offset that should be used for conversions from DateTime to DateTimeOffset.

Although this kind of "by convention" type rule based serialisation model may also add some interesting advantages.
I am aware that we can build our own formatters so if anything this is a nice way / feature request about improving the workflow for the dev.
Instead of defining a new formatter it would be nice to just tell the model "when dealing with this type format like this" as a simple model expression saving the need to write the formatter but I might be over-simplifying a more complex situation.
My thinking was tht I might a rule to say "do normal odata stuff but when you hit a T, handle it like this (t) => my custom expression that outs a string" but I guess that would need a dataformat param too in case the request was for xml instead of json (as an example).

In short ... I have plenty of ideas but am not sure how best I should solve my problem in the odata framework.

**Comment by phatcher** [comment]
@TehWardy You could take the other approach and tell OData that it's running under UTC e.g.

https://msdn.microsoft.com/en-us/library/system.web.odata.extensions.httpconfigurationextensions.settimezoneinfo(v=vs.118).aspx

Issue about returning local times from a server is that it's always wrong for someone, whereas always returning UTC means that clients can then handle local display issues

**Comment by TehWardy** [comment]
Nice catch @phatcher thanks :)

**Comment by phatcher** [comment]
Your welcome
